### ISSUE TITLE
Points not centered

### ISSUE THREAD

**Issue by davidanthoff** [vega/vega#1837]
When I display the following spec:
```
{
    "encoding": {
        "x": {
            "field": "a",
            "type": "quantitative"
        },
        "y": {
            "field": "b",
            "type": "quantitative"
        }
    },
    "data": {
        "values": [
            {
                "b": 2,
                "a": 2
            },
            {
                "b": 1,
                "a": 1
            }
        ]
    },
    "mark": "point"
}
```
And then zoom into it, I see that the point isn't really centered on the grid lines:

![image](https://user-images.githubusercontent.com/1036561/59323422-6ee19400-8c8f-11e9-99d8-af37119e08a7.png)

is a zoomed version of the middle point.

I'm using the SVG renderer, and then of course the same happens if I convert to PNG, PDF etc.

**Comment by domoritz** [comment]
This issue has been discussed previously in https://github.com/vega/vega-scenegraph/issues/41

**Comment by davidanthoff** [comment]
I shared a new alpha version of VegaLite.jl yesterday with some folks in the julia community and asked them to test it out with a simple plot and save it to a pdf, and every single one reported this issue back to me... People do notice it :)

Especially if one converts a small plot to a pdf, and then opens it, it is just _very_ obvious because the default zoom level of Acrobat is to enlarge the pdf to the screen size.

**Comment by domoritz** [comment]
Haha, yeah, it bugged me as well. Maybe we can revisit this issue @jheer given the feedback from the Julia testers.

**Comment by kanitw** [comment]
This [spec](https://vega.github.io/editor/#/url/vega/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6SgFY6AK1jcAdpAA04KABNkCAE6NCeHnJyQA4t27UANsgDksMLDzd4AazCFN8ZGG5V1YHamR0FJUh0VjtYRgAvZGwoQnRlQIhIOOVlRhlqGNFFJIB3RmUWGIAmAAZSnNxkRmp8EvLKyFsAT3MYyDdTU0SVdDxMHABtJQhQCHGoGXQ-dulWdTcAfVKeich57v1lPpFbewdYOnhYClXxyAAzbnUMPBjgSDxmwmj9Y9P5WPR1WFewB+2eD+kEB0QAvp8QchTIxUMxkOp2vJIBCRuc8Op0DJYFcbjFhhMJg8ni92hdGKYgYjIcgAB4udqA1ioOiwZqodjcUwAXl5FgMAHkBQYLCjKoT-miJY9nsDcahWKZMOKJVB0LB2qhZCxFqCztK6Qytn1kL5kAAKCpgLUyFjmpmoQYAHRBJpdAF0AJSfACMnytAf9nsgUvGqNVkojSRJcuuCqV+sJwQ1+htOtBi2QMgSKoN9MRxqBZstnzTiHtfWZztdQI9nrAAGowH7m-622AA8HQxBw6qxlGZaT9OhqNR1MIgYm1mPuKxCOxmvjIGXdSaFMvtYhV0DM9nIO7c0nuMQl34sfvD2tydDlCmwINkq5ogfu+d1afkFjFi5GG596-ewlfsI0HOUKSpKdzkNAtejwKsXRXUEPTAABCXkwBkRVTDAAAyHDUMYWAADl0CIis4MdBDN23ZA6xDCMwVDd00UYiAXySDB1AOfE0WA84phmVNvm4w9QPacd4DuUSWjaIZIHYb4L1DS51G4VB7hrAYoDmBZkGWMVlKzeBuFUe5X3WQg9RwPjpXJLoNIodBTFYYEAGIABZ4AAdgADnQHyDKjKA7C5HRCDMoKknCagpk2KBgBdUEwHNMtPRdHAzQAMTjPpyPgjdbS3JCLzACwAFJ2Asb0wBdABZT8ZA8C5nCfdKwHlXKHWrM8ZG-VqSpdF1PTBeio0A1VIFpDSECc4Epsha9TASVNqL1SFuAuC5fjuHBinG6VaWKab4Fm9p5qgRblqgRCTV3HMoA2rbTRiABaPbL3ORdrKaE7ZKgRcFsYG9NQavrf3BD6kmaI7vpmv7IABqBHOcv5SlY1V0YmTGwHR9jtN+tQeKJZSBLmxMxP0M1yeUNT0HSDTLrvQYAR2WZZ10-TAeBlbCposUWf6Nn5iWFYuaWzVVturMEjBd19qgTEMj+QYrQeaLYvafJCkQFE8bWFI0gyLIWJVGzJmmYEAeUmN2hhGRP2pZSaYwenvsBLSmnZkWFAuoHxdTUGfz-eXIEV6hlbVmoNf0FAajqCEOz1-jwZiDEXNEqJVNT9QXJYpQ8eCWlCaGXjlLhsnROuIHbXaTkCDU8mxwKGILic35RJ0PB4aSlLN2DUSlXYaFMuc2AdYynPkAH9Ah9MAUvCVcKJ-TwyZGM0zrPMwfoTvM21jnKz-keOkdqP9WnPaLKcrwPKWWR9PSoqqqUWxrHQxDiJ0lUKacB9E3S+lOXM65Mq5ZlPvJbg9d1KiSbsoAAygTdoVtpSwOzivaUzs6Z6Ham3Keylt6mDvK3QheDpQYFpAAUVpECGuOAKjKThDIKhNDT70OlDoRwRDcEZy-nSGIpR-7E0AYg-QyCjyaDAbbZAFwpLKVQTgYh7drbMHhvVLETUWop2nrPeeCJF5oNIUmDhDgADCs5aFnyjhfN41RTDmljrUPApB3KlGDB-XhP9myCPGGbH6p1REgIkRYyA5gZGN00PAkRUB5pyIiQYjuv5THmNPpHGK1ioBuApPY6ojjnGuMCkmTBrscEkO0TvFu3CGHoEodQyRdDRKMOYXUjsCTOEVNKcpT+2Y+F0LzmxEAYIgA) shows another example that the grid is a bit off. 


![image](https://user-images.githubusercontent.com/111269/59652152-5830b680-9141-11e9-9f67-d1c17f7f9713.png)

**Comment by jheer** [comment]
@davidanthoff Thanks for sharing. This is a known, long-standing issue as @domoritz mentions. We just don't (yet) happen to have a clean solution that doesn't introduce other problems elsewhere. The issue is particularly salient with scalable vector graphics because when you zoom in, what used to be 0.5 pixel offset gets magnified. I'd certainly love to find a solution.

@kanitw what is the problem with the image you shared? Is it that the bars are not perfectly centered? (Keep in mind the time span for each bar is inclusive of its left gridline but not the right gridline...)

**Comment by kanitw** [comment]
> @kanitw what is the problem with the image you shared? Is it that the bars are not perfectly centered? (Keep in mind the time span for each bar is inclusive of its left gridline but not the right gridline...)

Yep. Perhaps the fix for the time unit bar case is to add default "offset" on one side of the bar, akin to the 1 pixel offset we add to histogram by default.

**Comment by jheer** [comment]
@kanitw: That sounds reasonable for many cases, but likely fails as a general rule if the spans get much larger or smaller. In your case the gridlines are stroked (centered between pixels by default) and the bars are filled (pixel aligned by default), so I believe our offset is actually making the alignment better (not worse), by snapping lines to proper pixels. I think the lack of centering arises due to the left-inclusive / right-exclusive span, which should be mathematically correct.

I'll keep the issue focused on the point / grid alignment issue @davidanthoff raises. In the worst case, I could imagine adding config to control the offset, which at least might be useful for exported images, etc.
